lib/thread.c

```diff
@@ -498,6 +498,43 @@ DEFUN (clear_thread_cpu,
 	return CMD_SUCCESS;
 }
 
+static void show_thread_timers_helper(struct vty *vty, struct thread_master *m)
```

Comment:
1. Can this be used to get timers for threads in BGP or any other process ?
2. if_zebra_speed_update: is this the thread function that gets executed after the next pop up ? Do you think printing thread id or name etc. is more useful ?
3.  Can you consider to add a heading on top of time values as well ? For example 
```
Showing timers for default              next popup interval
--------------------------              --------------------
  rtadv_timer                                      00:00:00.520
```
4. code looks fine.

Reply:
1) Did you read the code in vtysh/vtysh.c?
2) What is a thread id?  I've never heard of it in FRR.  The if_zebra_speed_update is the name.  So I am really confused about what you are talking about here.
3) I think it's pretty self explanatory since it is mainly going to be used by developers for debugging.

Reply:
1. Nothing recently. Spent some time very long time back. 
2. I use thread::get_id() from pthread to print thread_id in my dedugging. This is in C++, and I assume C may also have similar API. I meant that. Not sure if we use this in FRR. The reason, I was checking, as we have multiple entries with same name in the above CLI output.
3. ..

Reply:
1) I suggest you spend a few second looking at how the code change in vtysh/vtysh.c allows this to work across all daemons.
2) Let's remember here this is an event system timer that happens to be called a thread.  There is no thread id and it makes no sense in the context of the FRR system events to have a thread id.  Besides if you look at the show_thread_timers_helper we clearly print out timers associated per pthread.